python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py

```diff
@@ -24,152 +26,98 @@ class VxlanInterfaceMixin(Protocol):
     """
 
     @cached_property
-    def vxlan_interface(self: AvdStructuredConfigNetworkServicesProtocol) -> dict | None:
+    def _multi_vtep(self: AvdStructuredConfigNetworkServicesProtocol) -> bool:
+        return self.shared_utils.mlag is True and self.shared_utils.evpn_multicast is True
+
+    @structured_config_contributor
+    def vxlan_interface(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
         """
-        Returns structured config for vxlan_interface.
+        Update structured config for vxlan_interface.
 
         Only used for VTEPs and for WAN
 
         This function also detects duplicate VNIs and raise an error in case of duplicates between
         all Network Services deployed on this device.
         """
         if not (self.shared_utils.overlay_vtep or self.shared_utils.is_wan_router):
-            return None
+            return
 
-        vxlan = {
-            "udp_port": 4789,
-        }
+        vxlan = EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan(udp_port=4789)
 
         if self._multi_vtep:
-            vxlan["source_interface"] = "Loopback0"
-            vxlan["mlag_source_interface"] = self.shared_utils.vtep_loopback
+            vxlan.source_interface = "Loopback0"
+            vxlan.mlag_source_interface = self.shared_utils.vtep_loopback
         else:
-            vxlan["source_interface"] = self.shared_utils.vtep_loopback
+            vxlan.source_interface = self.shared_utils.vtep_loopback
 
         if self.shared_utils.mlag_l3 and self.shared_utils.network_services_l3 and self.shared_utils.overlay_evpn:
-            vxlan["virtual_router_encapsulation_mac_address"] = "mlag-system-id"
+            vxlan.virtual_router_encapsulation_mac_address = "mlag-system-id"
 
         if self.shared_utils.overlay_her and not self.inputs.overlay_her_flood_list_per_vni and (common := self._overlay_her_flood_lists.get("common")):
-            vxlan["flood_vteps"] = natural_sort(unique(common))
+            vxlan.flood_vteps = EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan.FloodVteps(natural_sort(unique(common)))
 
         if self.shared_utils.overlay_cvx:
-            vxlan["controller_client"] = {"enabled": True}
-
-        vlans: list[dict] = []
-        vrfs: list[dict] = []
-        # vnis is a list of dicts only used for duplication checks across multiple types of objects all having "vni" as a key.
-        vnis: list[dict] = []
+            vxlan.controller_client = EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan.ControllerClient(enabled=True)
 
+        # TODO: maybe this could be made a class attribute?
```

Comment:
It makes sense to make it a class attribute if you set structured config directly in various helper functions.